src/util/strutil.c

```diff
@@ -17,8 +30,14 @@ void str_tolower(const char *str, char *lower, size_t *lower_len) {
 	//Update lower len
 	*lower_len = str_len;
 
+	utf8proc_int32_t c;
 	size_t i = 0;
-	for(; i < str_len; i++) lower[i] = tolower(str[i]);
+	int w = 0;
+	while(i < str_len) {
+		w = utf8proc_iterate(str + i, -1, &c);
+		utf8proc_encode_char(utf8proc_tolower(c), lower + i);
+		i+=w;
+	}
```

Comment:
can't we encode the entire string in one go (single function call) and then follow up with a single call to `to lower`?
I just hate to see this loop here...

Reply:
we can't